shared/utility/strategy/soccer_strategy.hpp

```diff
@@ -0,0 +1,124 @@
+        std::vector<Robots> robots;
+
+        // Transforms the ball's position from world coordinates to field coordinates using field.Hfw.
+        // rBWw is the ball's position in world coordinates.
+        Eigen::Vector3d rBFf = field.Hfw * ball.rBWw;
+
+        // Looping through teammmates, for each teammate, a 'Robots' object is created.
+        // Robot's position (mate.rRFf) is retrieved, and distance from robot to ball is calculated using norm.
+        // Then is stored in 'distance_to_ball'
+        // Each 'Robots' object is added to the robots vector.
+        for (const auto& mate : teammates.teammates) {
+            Robots robot;
+            robot.id               = mate.id;
+            Eigen::Vector3d rRFf   = mate.rRFf;
+            robot.distance_to_ball = (rRFf - rBFf).norm();
+            robots.push_back(robot);
+        }
+
+        // Robot's are sorted by distance to the ball, if there is a tie, they are sorted by robot id.
+        std::sort(robots.begin(), robots.end(), [](const Robots& a, const Robots& b) {
+            return (a.distance_to_ball < b.distance_to_ball)
+                   || (a.distance_to_ball == b.distance_to_ball && a.id < b.id);
+        });
+
+        // Sorted list of robots is returned.
+        return robots;
+    }
+
+
+    Possession get_possession(const Ball& ball, const TeamMates& teammates, const Field& field, double threshold) {
+
+        // Function determines who has possession based on proximity and a threshold distance.
+        // First calls 'get_sorted_bots()' to get the list of robots sotered by distance to ball.
+        // If no robot's are found, it returns 'Possession::NONE'
+        std::vector<Robots> sortedRobots = get_sorted_bots(ball, teammates, field);
+        if (sortedRobots.empty()) {
+            return Possession{Possession::NONE};
+        }
+
+        // rBCc transforms the ball's position from world coordinates to camera coordinates using ball.Hcw.
+        // If the distance between the ball and robot is less than the threshold, it will assume the robot has possesion
+        // 'Possession::SELF' is returned
+        Eigen::Vector3d rBCc = ball.Hcw * ball.rBWw;
+
+        if (rBCc.norm() < threshold) {
+            return Possession{Possession::SELF};
+        }
+
+        // Function checks each teammate, calculates the distance from each teammate to the ball's position.
+        // If there is a teammate within the threshold, 'Possession::TEAMMATE' is returned.
+        Eigen::Vector3d rBFf = field.Hfw * ball.rBWw;
+        for (const auto& teammate : teammates.teammates) {
+            Eigen::Vector3d rRBf = teammate.rRFf - rBFf;
+            if (rRBf.norm() < threshold) {
+                return Possession{Possession::TEAMMATE};
+            }
+        }
+
+        // If no robot or teammate id within the threshold, 'Possession::NONE' is returned.
+        return Possession{Possession::NONE};
```

Comment:
rather than do all this, get the first sorted robot (pseudocode below)
```
closest = sorted_robots[0];
if closest.distance is not within threshold then return Possession NONE
else if closest.id == me then return Possession{Possession::SELF};
else if closest.id == teammate then return Possession{Possession::TEAMMATE};
```
this will also require that you add the self robot into the sorting function